File: dixit-companion/src/main/java/fr/erban/dxitcompanion/db/game/GameConverter.java
package fr.erban.dxitcompanion.db.game;

import com.google.gson.Gson;

import fr.erban.dxitcompanion.game.GameBean;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class GameConverter {

    public static GameEntity toEntity(final GameBean gameBean) {

        return GameEntity.builder()
                .finished(gameBean.isFinished())
                .nbTurns(gameBean.getCurrentTurn())
                .nameWinner(gameBean.getNameWinner())
                .pointsToWin(gameBean.getPointsToWin())
                .scoreSheet(gameBean.getScoresheet() != null ? new Gson().toJson(gameBean.getScoresheet()) : null)
                .build();
    }
}

File: dixit-companion/src/main/java/fr/erban/dxitcompanion/db/game/GameConverter.kt
package fr.erban.dxitcompanion.db.game

import com.google.gson.Gson
import fr.erban.dxitcompanion.game.GameBean

object GameConverter {

    fun toEntity(gameBean: GameBean): GameEntity = GameEntity(
        finished = gameBean.finished,
        nbTurns = gameBean.currentTurn,
        nameWinner = gameBean.nameWinner,
        pointsToWin = gameBean.pointsToWin,
        scoreSheet = Gson().toJson(gameBean.getScoresheet())
    )
}
